server/gc/src/test/java/org/apache/accumulo/gc/GarbageCollectionTest.java

```diff
@@ -122,6 +158,28 @@ public void incrementInUseStat(long i) {}
     public Iterator<Entry<String,Status>> getReplicationNeededIterator() {
       return filesToReplicate.entrySet().iterator();
     }
+
+    @Override
+    public Set<TableId> getCandidateTableIDs() {
+      if (level == Ample.DataLevel.ROOT) {
+        return Set.of(RootTable.ID);
+      } else if (level == Ample.DataLevel.METADATA) {
+        return Collections.singleton(MetadataTable.ID);
+      } else if (level == Ample.DataLevel.USER) {
+        tableIds.remove(MetadataTable.ID);
+        tableIds.remove(RootTable.ID);
+        getTableIDs().forEach((k, v) -> {
+          if (v == TableState.ONLINE || v == TableState.OFFLINE) {
+            // Don't return tables that are NEW, DELETING, or in an
+            // UNKNOWN state.
+            tableIds.add(k);
+          }
+        });
+        return tableIds;
```

Comment:
It would be good to avoid modifying the instance var tableIds.

```suggestion
        Set<TableId> tableIds = new HashSet<>();
        getTableIDs().forEach((k, v) -> {
          if (v == TableState.ONLINE || v == TableState.OFFLINE) {
            // Don't return tables that are NEW, DELETING, or in an
            // UNKNOWN state.
            tableIds.add(k);
          }
        });
        tableIds.remove(MetadataTable.ID);
        tableIds.remove(RootTable.ID);
        return tableIds;
```

Reply:
Fixed.